### PR TITLE
common/TrackedOp: do not count the ops marked as nowarn

### PR DESCRIPTION
**Pull request by lxbsz**
If an op is marked as nowarn then it won't be counted as the slow requests, but currently it will count the initiated time when iterating the inflight ops.

For example:

[WRN] : 1 slow requests, 1 included below; oldest blocked for > 38.764892 secs
[WRN] : slow request 33.875059 seconds old, received at 2024-06-17T14:14:34.228261+0000: client_request(client.78109915:11369251 mkdir #0x1008ecedea2/chk-89588 2024-06-17T14:14:34.097825+0000 caller_uid=1002960000, caller_gid=0{0,1002960000,}) currently failed to wrlock, waiting

The oldest blocked request is 38.764892 old, but the oldest slow request reported is 33.875059 old.

This will fix the old commit from https://github.com/ceph/ceph/pull/21715.

Fixes: https://tracker.ceph.com/issues/66557

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by leonid-s-usov** [comment]
@batrick we had to introduce a new field `continuous` in the quiesce epic to achieve the same, right? Should we review?

File: src/common/TrackedOp.cc [hunk 1]
```diff
@@ -339,12 +339,15 @@ bool OpTracker::visit_ops_in_flight(utime_t* oldest_secs,
   for (const auto sdata : sharded_in_flight_list) {
     ceph_assert(sdata);
     std::lock_guard locker(sdata->ops_in_flight_lock_sharded);
-    if (!sdata->ops_in_flight_sharded.empty()) {
-      utime_t oldest_op_tmp =
-	sdata->ops_in_flight_sharded.front().get_initiated();
+    for (auto& op : sdata->ops_in_flight_sharded) {
+      if (!op.warn_interval_multiplier)
```
Comment (batrick) [inline src/common/TrackedOp.cc:343]:
```suggestion
      if (!op.warn_interval_multiplier || op->is_continuous())
```

**Review by batrick — CHANGES_REQUESTED** [review]
otherwise lgtm

**Comment by batrick** [comment]
> @batrick we had to introduce a new field `continuous` in the quiesce epic to achieve the same, right? Should we review?

We could consolidate them, yes.

File: src/common/TrackedOp.cc (hunk 1 above)
Comment (lxbsz) [reply src/common/TrackedOp.cc:343]: Good catch. Will fix it.

**Review by vshankar — APPROVED** [review]

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/66609](https://tracker.ceph.com/issues/66609).

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
https://tracker.ceph.com/issues/66609#note-1

**Pull request merged**